### PR TITLE
docs: show a warning banner for the latest docs (readthedocs+mkdocs)

### PR DESCRIPTION
**Pull request by stasadev**
## The Issue

- #5832

## Alternative

- #5859

## How This PR Solves The Issue

Adds a warning banner.

Resources used in making this:
- https://docs.readthedocs.io/en/stable/reference/environment-variables.html#envvar-READTHEDOCS_VERSION_NAME
- https://docs.readthedocs.io/en/stable/versions.html#version-warning
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-versioning/?h=outdated#version-warning
- https://github.com/readthedocs/readthedocs.org/issues/8529

Fixes mkdocs warnings on build.

## Manual Testing Instructions

See https://ddev--5855.org.readthedocs.build/en/5855/

image

## Alternative

As an alternative we could use this https://docs.readthedocs.io/en/stable/versions.html#version-warning

image

But it appears over and over again when you navigate the latest docs, so I decided not to use it (disabled it here https://beta.readthedocs.org/dashboard/ddev/addons/edit/).

### CODE REVIEW

**Comment by stasadev** [comment]
@bmartinez287 please review this 🙂

**Comment by rfay** [comment]
Oh fantastic!

**Comment by bmartinez287** [comment]
Looks good, I am testing with an alternative with the announcement block bar.
https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/#announcement-bar.
https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks
We might be able to use an environment variable on the mkdocs.yml file to trigger that block.
It be hard to avoid the flickering all together since the javascript will always load a little bit later.

File: docs/content/assets/warning-banner-for-latest.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+function load_warning_banner_for_latest() {
+    let currentUrl = window.location.href;
+
+    if (currentUrl.includes('/latest/')) {
+        // Insert the warning banner as the first element in the body
+        document.body.insertAdjacentHTML('afterbegin', `
+            <div id="warning-banner-for-latest" data-md-color-scheme="default" style="display: none;">
+                <aside class="md-banner md-banner--warning">
+                    <div class="md-banner__inner md-grid md-typeset">
+                        You’re viewing the latest unreleased version.
```
Comment (GuySartorelli) [inline docs/content/assets/warning-banner-for-latest.js:14]:
```suggestion
                        You're viewing documentation for an unreleased version of DDEV.
```

File: docs/content/assets/warning-banner-for-latest.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                        <a href="${currentUrl.replace(/\/latest\//, '/stable/')}">
+                            <strong>Click here to go to stable.</strong>
```
Comment (GuySartorelli) [inline docs/content/assets/warning-banner-for-latest.js:16]:
```suggestion
                            <strong>Click here to see the stable documentation.</strong>
```

**Review by GuySartorelli — CHANGES_REQUESTED** [review]
That's great!

The below suggested change makes the language a little more explicit,  e.g. removing "latest" and emphasising "unreleased", since some people might skim read and see "latest" and think "well good, im using the latest version" without realising the specific meaning here that "latest" implies "unreleased"

**Comment by rfay** [comment]
Rebased to get docs test passing.

**Comment by stasadev** [comment]
I made it work for PRs as well and moved the warning banner, so now the menu doesn't jump, only the content:

https://ddev--5855.org.readthedocs.build/en/5855/

**Comment by bmartinez287** [comment]
Pretty cool, that is better.

**Review by GuySartorelli — APPROVED** [review]

**Comment by stasadev** [comment]
I made it work as it should. No more flickering 🎉

**Comment by bmartinez287** [comment]
Looks great.

File: docs/content/developers/maintainers.md [hunk 3]
```diff
@@ -46,7 +46,8 @@ There are cases like access to hosting provider integrations that have essential
 * **CircleCI**: Maintainers automatically have some access via their GitHub team membership, but should probably get more.
 * **developer.apple.com**: Add to the DDEV team there so certificates can be managed.
 * **hub.docker.com**: Add user to owners team in DDEV org.
-* **Chocolatey**: Add user to [Manage maintainers](https://community.chocolatey.org/packages/ddev/1.22.5/ManagePackageOwners).
+* **Chocolatey**: Add user to [Manage maintainers](https://community.chocolatey.org/packages/ddev/ManagePackageOwners).
+* **Read the Docs**: Add user to [Maintainers](https://readthedocs.org/dashboard/ddev/users/).
```
Comment (rfay) [inline docs/content/developers/maintainers.md:50]: Yay!

File: docs/content/index.md [hunk 4]
```diff
@@ -62,7 +62,7 @@ These environments can be extended, version controlled, and shared, so you can t
     
     1. Install Docker with [recommended settings](users/install/docker-installation.md#windows).
     2. Install [DDEV for Windows](users/install/ddev-installation.md#windows).
-    3. Launch your [first project](users/project) and start developing. 🚀
+    3. Launch your [first project](users/project.md) and start developing. 🚀
```
Comment (rfay) [inline docs/content/index.md:65]: I wonder why these weren't caught by links check.

**Review by rfay — APPROVED** [review]
Thanks!

**Comment by rfay** [comment]
@mattstein do you want to take a look?

**Review by bmartinez287 — APPROVED** [review]

File: docs/content/index.md (hunk 4 above)
Comment (stasadev) [reply docs/content/index.md:65]: I saw the warning in the build only after I fixed the deprecated emoji plugins 🙂.
The link itself works as before (mkdocs gives us some slack).

**Pull request merged**